addOns/client/src/main/java/org/zaproxy/addon/client/spider/MessagesTable.java

```diff
@@ -0,0 +1,151 @@
+import org.zaproxy.addon.client.spider.ClientSpider.ResourceState;
+import org.zaproxy.addon.client.spider.MessagesTableModel.ProcessedCellItem;
+import org.zaproxy.zap.utils.DisplayUtils;
+import org.zaproxy.zap.view.table.HistoryReferencesTable;
+
+@SuppressWarnings("serial")
+public class MessagesTable extends HistoryReferencesTable {
+
+    private static final long serialVersionUID = 1L;
+
+    private static final String RESULTS_TABLE_NAME = "ClientSpiderMessagesTable";
+
+    private final ExtensionHistory extensionHistory;
+
+    public MessagesTable(MessagesTableModel model) {
+        super(model);
+
+        setName(RESULTS_TABLE_NAME);
+
+        setAutoCreateColumnsFromModel(false);
+
+        getColumnExt(0)
+                .setCellRenderer(
+                        new DefaultTableRenderer(
+                                new MappedValue(StringValues.EMPTY, IconValues.NONE),
+                                JLabel.CENTER));
+        getColumnExt(0).setHighlighters(new ProcessedCellItemIconHighlighter(0));
+
+        getColumnExt(Constant.messages.getString("view.href.table.header.timestamp.response"))
+                .setVisible(false);
+        getColumnExt(Constant.messages.getString("view.href.table.header.size.requestheader"))
+                .setVisible(false);
+        getColumnExt(Constant.messages.getString("view.href.table.header.size.requestbody"))
+                .setVisible(false);
+
+        extensionHistory =
+                Control.getSingleton().getExtensionLoader().getExtension(ExtensionHistory.class);
+    }
+
+    @Override
+    public void setModel(TableModel dataModel) {
+        // Keep the same column sorted when model is changed
+        int sortedcolumnIndex = getSortedColumnIndex();
+        SortOrder sortOrder = getSortOrder(sortedcolumnIndex);
+        super.setModel(dataModel);
+        if (sortedcolumnIndex != -1) {
+            setSortOrder(sortedcolumnIndex, sortOrder);
+        }
+    }
+
+    @Override
+    protected HistoryReference getHistoryReferenceAtViewRow(int row) {
+        HistoryReference historyReference = super.getHistoryReferenceAtViewRow(row);
+        if (historyReference == null) {
+            return null;
+        }
+
+        if (extensionHistory == null
+                || extensionHistory.getHistoryReference(historyReference.getHistoryId()) == null) {
+            // Associated message was deleted in the meantime.
+            return null;
+        }
+
+        return historyReference;
+    }
+
+    private static class ProcessedCellItemIconHighlighter extends AbstractHighlighter {
+
+        private static final ImageIcon ALLOWED_ICON =
+                DisplayUtils.getScaledIcon(
+                        MessagesTable.class.getResource("/resource/icon/16/152.png"));
+
+        private static final ImageIcon NOT_ALLOWED_ICON =
+                DisplayUtils.getScaledIcon(
+                        MessagesTable.class.getResource("/resource/icon/16/149.png"));
```

Comment:
Could we give the icons useful filenames?

I know this is a holdover from code/history but I think it'll be more maintenance friendly in the long run.

Reply:
I guess these already exist, so maybe that's a change for another PR.

Reply:
Those are core icons.